### ISSUE TITLE
Integrated terminal doesn't toggle when it's losing focus

### ISSUE THREAD

**Issue by alekseyt** [Microsoft/vscode#61910]
Issue Type: <b>Bug</b>

Steps to reproduce:

1. Toggle integrated terminal to make it appear (Ctrl+`)

Note that it has keyboard focus.

![image](https://user-images.githubusercontent.com/993454/47595393-e379f380-d988-11e8-954f-3633fc45a547.png)

2. Perform some action to make it lose focus. Options: start a build job when you have more than one (so their list appear for selection), press Ctrl-P, or open some other "popup-window" e.g. open "Goto to symbol" quicklist (or how is it called), right clicking in editor also reproduces this.

Was initially reproduced when starting build jobs, but reproduces on any window poping up.

Actual result 1: terminal loses focus

![image](https://user-images.githubusercontent.com/993454/47595418-f391d300-d988-11e8-87c1-be134b3603ad.png)

Actual result 2: now you have to press Ctrl+` twice to toggle terminal. First time it's getting focus, second time it hides (actually toggles).

I can not reproduce this in 1.27.2

VS Code version: Code 1.28.2 (7f3ce96ff4729c91352ae6def877e59c561f4850, 2018-10-17T00:20:56.183Z)
OS version: Linux x64 4.4.0-112-generic

<details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
cpptools|ms-|0.19.0
shader|sle|1.1.4
vscode-lua|tri|0.12.4
cmake|twx|0.0.17
vscode-todo-highlight|way|1.0.4

(1 theme extensions excluded)

</details>

**Comment by usernamehw** [comment]
https://code.visualstudio.com/updates/v1_28#_toggle-panel-changes

**Comment by alekseyt** [comment]
That explains it. I've read #19400 and the actual workaround seems to be to rebind Ctrl+` to "Toggle Panel". This is not quite "Toggle terminal", but OK.

However the problem persists: when you have several build jobs and you start them one by one or several jobs in parallel, terminal loses focus and you need to hit Ctrl+` twice to hide it. This is hard to use when i have to remember or pay attention in what situations i need to hit Ctrl+` twice or once. 

Maybe something could be done to alleviate this i don't know, maybe return focus to open terminal when quicklist menu closes. I suspect this might also solve the issue without the need to change defaults or new behavior. Why focus changes to the editor instead of open terminal where it was before, does that work as intended?

I have this ever growing list of changed defaults and this entire configuration also hard to maintain (transfer across different virtual machines, for instance, alternative is to reconfigure every installation of vscode). I would love to use defaults or a default preset of settings if that matters, and would rather adapt to defaults to avoid maintaining "dotfiles", but sometimes i don't have another choice but to change defaults.

By the way i don't see how this is consistent to sidebar behavior, sidebar toggles even when it doesn't have focus and this is exactly what i would expect it to do.

**Comment by alekseyt** [comment]
Actually this works OK for me with terminal hotkey rebound to `Toggle Panel`, so i'm closing this issue. This is not the most important issue with integrated terminal, is it? I suppose someone will file another issue if this is important.
